networkServer/src/main/java/org/pesc/cds/service/OrganizationService.java

```diff
@@ -0,0 +1,101 @@
+package org.pesc.cds.service;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.pesc.cds.config.CacheConfig;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+
+/**
+ * Created by sallen on 8/12/16.
+ */
+@Service
+public class OrganizationService {
+    private static final Log log = LogFactory.getLog(OrganizationService.class);
+    private static final String ORGANIZATION = CacheConfig.PREFIX + "Organization";
+    private static final String ORGANIZATIONS = CacheConfig.PREFIX + "Organizations";
+
+    @Value("${directory.server.base.url}")
+    private String directoryServer;
+
+    @Value("${api.organization}")
+    private String organizationApiPath;
+
+    @Cacheable(ORGANIZATION)
+    public JSONObject getOrganization(Integer organizationId){
+        return getOrganization(organizationId, null, null);
+    }
+
+    public JSONObject getOrganization(String destinationSchoolCode, String destinationSchoolCodeType){
+        return getOrganization(null, destinationSchoolCode, destinationSchoolCodeType);
+    }
+
+    private JSONObject getOrganization(Integer organizationId, String destinationSchoolCode, String destinationSchoolCodeType){
+        JSONObject organization = null;
+        StringBuilder uri = new StringBuilder(directoryServer + organizationApiPath);
+        if(organizationId!=null) {
+            uri.append("?id=").append(organizationId);
+        }
+        if(destinationSchoolCode!=null && destinationSchoolCodeType!=null){
+            uri.append("?organizationCodeType=").append(destinationSchoolCodeType).append("&organizationCode=").append(destinationSchoolCode);
+        }
+        CloseableHttpClient client = HttpClients.custom().build();
+        try {
```

Comment:
I'm only saying this because it's a construct I like, and this is new code, and I'm assuming Java 7 or above. If CloseableHttpClient and ClosableHttpResponse below are true implementers of the Closeable interface, then this is much cleaner IMO to use the try (CloseableHttpClient client = HttpClients.custom().build()) { ... } and simply omitting the finally blocks closing (and also catching)